src/codegen/llvm.zig

```diff
@@ -8622,6 +8644,26 @@ pub const FuncGen = struct {
         return null;
     }
 
+    fn safeWasmMemset(
+        self: *FuncGen,
+        dest_ptr: *llvm.Value,
+        fill_byte: *llvm.Value,
+        len: *llvm.Value,
+        dest_ptr_align: u32,
+        is_volatile: bool,
+    ) !void {
+        const parent_block = self.context.createBasicBlock("Block");
+        const llvm_usize_ty = self.context.intType(self.dg.object.target.ptrBitWidth());
+        const cond = try self.cmp(len, llvm_usize_ty.constInt(0, .False), Type.usize, .neq);
+        const then_block = self.context.appendBasicBlock(self.llvm_func, "Then");
+        _ = self.builder.buildCondBr(cond, then_block, parent_block);
+        self.builder.positionBuilderAtEnd(then_block);
+        _ = self.builder.buildMemSet(dest_ptr, fill_byte, len, dest_ptr_align, is_volatile);
+        _ = self.builder.buildBr(parent_block);
+        self.llvm_func.appendExistingBasicBlock(parent_block);
+        self.builder.positionBuilderAtEnd(parent_block);
```

Comment:
```suggestion
        const llvm_usize_ty = self.context.intType(self.dg.object.target.ptrBitWidth());
        const cond = try self.cmp(len, llvm_usize_ty.constInt(0, .False), Type.usize, .neq);
        const memset_block = self.context.appendBasicBlock(self.llvm_func, "MemsetTrapSkip");
        const end_block = self.context.appendBasicBlock(self.llvm_func, "MemsetTrapEnd");
        _ = self.builder.buildCondBr(cond, memset_block, end_block);
        self.builder.positionBuilderAtEnd(memset_block);
        _ = self.builder.buildMemSet(dest_ptr, fill_byte, len, dest_ptr_align, is_volatile);
        _ = self.builder.buildBr(end_block);
        self.builder.positionBuilderAtEnd(end_block);
```

Reply:
Aha, I get it now, thanks! Updated both functions.

Reply:
No problem! I think your wasm expertise was actually getting in the way, because you knew how the code *eventually* had to be lowered! But LLVM wants to operate on basic blocks before it lowers into wasm's nested block style.